voidcrew/code/modules/overmap/ships/pvp/subverter.dm

```diff
@@ -0,0 +1,329 @@
+//Used in subvert_bark
+#define SUBVERTER_BARK_FAILURE 'sound/machines/buzz-two.ogg'
+#define SUBVERTER_BARK_SUCCESS 'sound/machines/ping.ogg'
+//Default value for how long the subverter must recharge between uses
+#define SUBVERTER_RECHARGE_TIME (2 MINUTES)
+//Default value for how long a subversion target cannot control the ship
+#define SUBVERTER_ENGINE_STALL_TIME (1 MINUTES)
+//How hot the subverter makes the room when used
+#define SUBVERTER_SPICINESS 150
+/**
+* These get returned by can_subvert, and are mostly used to make the helm console say different things when a subversion is attempted
+*/
+#define SUB_SUCCESS 0
+#define SUB_TARGET_IS_NEU 1
+#define SUB_WE_ARE_NEU 2
+#define SUB_TARGET_IS_ALLY 3
+#define SUB_RECHARGING 4
+#define SUB_TARGET_SUBVERTED 5
+#define SUB_TARGET_DOCKED 6
+#define SUB_TARGET_DOCKING 7
+#define SUB_TARGET_UNDOCKING 8
+#define SUB_TARGET_GRACE 9
+#define SUB_TARGET_ANTIVIRUS 10
+
+/obj/machinery/subverter
+	name = "interdictor"
+	desc = "A piece of equipment used to upload programs to other vessels. It has no interface: It could be linked to an auxiliary console or the wires could be pulsed manually."
+	icon = 'voidcrew/icons/obj/machines/space_hacking.dmi'
+	icon_state = "subverter"
+	density = TRUE
+	use_power = IDLE_POWER_USE
+	idle_power_usage = 100
+	active_power_usage = 1000
+	circuit = /obj/item/circuitboard/machine/subverter
+	layer = BELOW_OBJ_LAYER
+	//Which ship the subverter is on
+	var/obj/structure/overmap/ship/simulated/ship
+	//Linked auxiliary console
+	var/obj/machinery/computer/autopilot/aux
+	//Standard machine stuff
+	var/hacked = FALSE
+	var/disabled = FALSE
+	var/shocked = FALSE
+	//Cooldown that handles the subverter recharging
+	COOLDOWN_DECLARE(subverter_cooldown)
+	//Used to handle higher tier parts affecting the subverter
+	var/sub_recharge = SUBVERTER_RECHARGE_TIME
+	var/sub_engine = SUBVERTER_ENGINE_STALL_TIME
+	var/spiciness = SUBVERTER_SPICINESS
+
+/obj/machinery/subverter/Initialize()
+	. = ..()
+	wires = new /datum/wires/subverter(src)
+	update_stats()
+	COOLDOWN_START(src, subverter_cooldown, sub_recharge)
+
+/obj/machinery/subverter/connect_to_shuttle(obj/docking_port/mobile/port, obj/docking_port/stationary/dock)
+	ship = port.current_ship
+
+/obj/machinery/subverter/on_deconstruction()
+	if (aux)
+		aux.sub = null
+		aux = null
+	..()
+
+/**
+*	Returns formatted remaining time till the subverter finishes recharging
+*/
+/obj/machinery/subverter/proc/get_recharge_str()
+	. = COOLDOWN_TIMELEFT(src, subverter_cooldown) / 10
+	. *= . > 0
```

Comment:
Whats the point of this?
if `.` is 0, then multiply it by zero, otherwise multiply by one? this doesn't seem necessary

Reply:
oh ye thats leftover from when i didnt use the macros